picoquic/spinbit.c

```diff
@@ -77,6 +79,26 @@ uint8_t picoquic_spinbit_vec_outgoing(picoquic_cnx_t * cnx)
 	return spin_vec | (uint8_t)((cnx->path[0]->current_spin) << 2);
 }
 
+/*
+ * Two procedures defining the spin bit with QR loss bits
+ */
+
+void picoquic_spinbit_sqr_incoming(picoquic_cnx_t * cnx, picoquic_path_t * path_x, picoquic_packet_header * ph)
+{
+    path_x->current_spin = ph->spin ^ cnx->client_mode;
+}
+
+uint8_t picoquic_spinbit_sqr_outgoing(picoquic_cnx_t * cnx,uint32_t flags)
+{
+  picoquic_path_t *pa = cnx->path[0];
+  pa->loss_q_index++;
+  if (pa->loss_q_index>=PICOQUIC_LOSS_Q_PERIOD) {
+    pa->loss_q_index=0;
+    pa->loss_q=(1-pa->loss_q);
+  }
+  return (uint8_t)((pa->current_spin << 2)|(pa->loss_q << 1)|((flags&PICOQUIC_SPINFLAGS_RETRANS)!=0));
+}
```

Comment:
You can achieve the same result by having an element in the path context:
```
unsigned int loss_detected:1;
```
Then do:
```
uint8_t spinbyte_sqr = (uint8_t)((pa->current_spin << 2)|(pa->loss_q << 1)|(pa->loss_detected);
pa->loss_detected = 0;
return spinbyte_sqr;
```

Reply:
OK, done as suggested. The per-path  state is now a retrans_count, to help accurate reporting of the number of lost packets regardless of frame re-aggregation. It is also accounted for in the original lossy path, so that a middle observer has a consistent, local view.